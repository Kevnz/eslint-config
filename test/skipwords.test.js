const eslintConfig = require('../eslint-config.json')
const skipWords = eslintConfig.rules['spellcheck/spell-checker'][1].skipWords

function hasDuplicates(arr) {
  let x = {}
  let length = arr.length

  for (let i = 0; i < length; i++) {
    if (x[arr[i]] === true) {
      return true
    }
    x[arr[i]] = true
  }
  return false
}
describe('The Spelling rules', () => {
  it('should not change', () => {
    expect(skipWords).toMatchSnapshot()
  })
  it('should not have duplicates', () => {
    expect(hasDuplicates(skipWords)).toBe(false)
  })
})
