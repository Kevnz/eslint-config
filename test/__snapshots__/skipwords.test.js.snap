// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Spelling rules should not change 1`] = `
Array [
  "aff",
  "api",
  "apollo",
  "args",
  "async",
  "blipp",
  "cors",
  "defs",
  "del",
  "dev",
  "dict",
  "dom",
  "ejs",
  "err",
  "eslint",
  "filename",
  "github",
  "gql",
  "graphql",
  "hapi",
  "html",
  "http",
  "hunspell",
  "hunspellchecker",
  "inmemory",
  "jsx",
  "knex",
  "lodash",
  "middleware",
  "mjs",
  "resolvers",
  "src",
  "stderr",
  "stdin",
  "stdout",
  "uri",
  "util",
  "utils",
  "webpack",
  "xtconf",
]
`;
